src/flake8_simplify/plugins/ast_bool_op.rs

```diff
@@ -143,6 +145,29 @@ pub fn duplicate_isinstance_call(checker: &mut Checker, expr: &Expr) {
     }
 }
 
+define_violation!(
+    pub struct AAndNotA(String);
+);
+
+impl Violation for AAndNotA {
+    fn message(&self) -> String {
+        let name = &self.0;
+        format!("Use `False` instead of `{name} and not {name}`")
+    }
+
+    fn autofix_available(&self) -> bool {
+        true
+    }
+
+    fn autofix_description(&self) -> Option<String> {
```

Comment:
I think ideally `autofix_available` and `autofix_description` would be coupled. We enforce it with a test now. (e.g., `autofix_available` is effectively `autofix_description.is_some()`).

Reply:
Thanks for bringing that up! I agree. I think the best solution is to replace both trait methods with:

```rs
    fn autofix_formatter(&self) -> Option<fn(&Self) -> String> { None }
```

Since implementing that trait method however becomes a bit awkward since you need to define the formatter as an additional function, so I think it would make sense to introduce another trait `ViolationWithAutofix` that implements `Violation` via a blanket implementation ... I already implemented that, I'll force push it shortly.